IPLD-Data-Model-v1.md

```diff
@@ -1,17 +1,108 @@
-# [WIP] IPLD Data Model
+IPLD Data Model
+===============
 
-## Simple Types
+The IPLD Data Model is a core part of the IPLD specification,
+which describes what data is representable in IPLD --
+for example, booleans, integers, textual strings, maps and lists, etc.
+
+While the Data Model describes these representations in the abstract,
+[Codecs](Codecs) specify exactly how these data are transcribed into serialized bytes.
+(Another component of the IPLD specifications, [Schemas](schema), provide
+additional optional tooling on top of the Data Model which can further refine,
+describe, and constrain the range of acceptable data values.)
+
+
+Kinds
+-----
+
+The following is the list of essential _kinds_ (or more formally, _representation kinds_)
+of data representable in the IPLD Data Model:
 
-* Boolean
 * Null
+* Boolean
+* Integer
+* Float
 * String
+* Bytes
+* List
+* Map
+* Link
+
+(Note that we use the term "_kinds_" here to disambiguate this from "_types_",
+which is a term we'll use at the [Schemas](schema) level.)
+
+The _recursive kinds_ are:
+
+* List
+* Map
+
+The _scalar kinds_ (the complement of recursive) are:
+
+* Null
+* Boolean
 * Integer
 * Float
-* Array
-* Object (Hash Map)
-* Binary
+* String
+* Bytes
+* Link
+
+(Note that [Schemas](schema) introduce a few more kinds -- when clarification is necessary,
+these Data Model kinds can be called the "_representation kinds_",
+while the additional kinds introduced in the Schema layer are "_perceived kinds_".)
```

Comment:
wait, aren't those "types"?

Reply:
each type is of some kind -- which may either be a representation kind or a perceived kind.

more on that in the other PR, but for example: a type may be of kind 'struct', which is a perceived kind (in that it doesn't exist at the Data Model layer) (and also needs to have a specified representation, which for a struct will probably be 'map' for example).